cmd/habitatctl/commands/community.go

```diff
@@ -102,59 +134,35 @@ var communityJoinCmd = &cobra.Command{
 			req.CommunityID = joinInfo.CommunityID
 		}
 
-		var res ctl.CommunityJoinResponse
-		postRequest(ctl.CommandCommunityJoin, req, &res)
-
-		fmt.Println(res.AddMemberToken)
-	},
-}
+		userIdentity, err := loadUserIdentity(cmd)
+		if err != nil {
+			printError(fmt.Errorf("error loading user identity: %s", err))
+		}
 
-var communityAddMemberCmd = &cobra.Command{
-	Use:   "add",
-	Short: "add a member to the community",
-	Run: func(cmd *cobra.Command, args []string) {
-		req := &ctl.CommunityAddMemberRequest{}
-		token, err := cmd.Flags().GetString("token")
+		conn, err := getWebsocketConn(ctl.CommandCommunityJoin)
 		if err != nil {
-			printError(err)
+			printError(fmt.Errorf("error establishing websocket connection: %s", err))
 		}
-		if token == "" {
-			address := cmd.Flags().Lookup("address")
-			if address == nil {
-				fmt.Println("address flag needs to be set")
-				return
-			}
+		defer conn.Close()
 
-			communityID := cmd.Flags().Lookup("community")
-			if communityID == nil {
-				fmt.Println("community flag needs to be set")
-				return
-			}
+		err = client.WebsocketKeySigningExchange(conn, userIdentity)
```

Comment:
same thing here about websocket. actually i feel like it would be really nice if we have code in identity or a separate package that separates all of this out

Reply:
yeah definitely one of the ickier things is that I have this key exchange code splite between the client and community packages. Could definitely be refactored in the future